src/playbacks/html5_video/html5_video.js

```diff
@@ -494,25 +495,33 @@ export default class HTML5Video extends Playback {
 
   setClosedCaptionsTrackId(trackId) {
     let tracks = this.closedCaptionsTracks
+    // Since it is possible to display multiple tracks,
+    // ensure that all tracks are hidden.
+    let showingTrack
     for (let i = 0; i < tracks.length; i++) {
       let track = tracks[i]
+      // Check if track to show
       if (trackId === track.id) {
-        if (track.track.mode === 'hidden' || track.track.mode === 'disabled') {
-          track.track.mode = 'showing'
-          this._ccTrackId = trackId
-        }
-      } else if (track.track.mode === 'showing') {
+        showingTrack = track
+      }
+      if (track.track.mode !== 'hidden') {
+        // In theory 'disabled' may be more optimized, but
+        // it leads to weird behaviours with some browsers.
+        // FIXME: value set could be browser specific ?
         track.track.mode = 'hidden'
       }
     }
```

Comment:
Here I would try to make it only a matter of transformations, filtering and select also.

```javascript
// since trackId is a parameter and it seems to be never changed, we can check it before all
if (!isNumber(trackId) || trackId < 0) {return}

showingTrack = tracks.select((track) => {
  if (track.id ==trackId) 
     return track
})

tracks.filter((track) => {track.track.mode !== 'hidden'})
           .forEach((track) => {track.track.mode = 'hidden'})

!!showingTrack && showingTrack.track.mode = 'showing'
this._ccTrackId = trackId
this.trigger(...)
```